modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1226,6 +1226,10 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
                 model.getVendorExtensions().put("x-jackson-optional-nullable-helpers", true);
             }
         }
+
+        if (property.isReadOnly) {
+            model.getVendorExtensions().put("hasReadonlyProperties", true);
```

Comment:
@borsch thanks for the PR.

For extensions, we're using the convention kebab-case with `x-` prefix.

e.g. x-jackson-optional-nullable-helpers

Reply:
@wing328 thanks for review. I have renamed extension variable to `x-has-readonly-properties`
please review